### PR TITLE
Add etcd server check at startup to kube2sky (with retry loop).

### PR DESCRIPTION
**Pull request by fabioy**
Addresses an issue with etcd not having started when kube2sky comes up.

### CODE REVIEW

File: cluster/addons/dns/kube2sky/kube2sky.go [hunk 1]
```diff
@@ -87,8 +88,23 @@ func mutateEtcdOrDie(mutator func() error) {
 }
 
 func newEtcdClient() (client *etcd.Client) {
-	// TODO: take a flag for etcd server(s).
-	client = etcd.NewClient([]string{"http://127.0.0.1:4001"})
+	maxConnectRetries := 12
+	for maxConnectRetries > 0 {
+		if _, _, err := tools.GetEtcdVersion(*etcd_server); err != nil {
```
Comment (thockin) [inline cluster/addons/dns/kube2sky/kube2sky.go]: where does "tools" come from - I see no pkg named tools?

**Comment by thockin** [comment]
LGTM, just one thing I don't understand :)

**Comment by thockin** [comment]
@rsokolowski - we should integrate this before cutting another container.

File: cluster/addons/dns/kube2sky/kube2sky.go (hunk 1 above)
Comment (fabioy) [inline cluster/addons/dns/kube2sky/kube2sky.go]: It's the tools package under "pkg/tools". It contained a number of nice etcd tools, which I modified and reused.

Comment (fabioy) [inline cluster/addons/dns/kube2sky/kube2sky.go]: Wait, I think a bad merge dropped its import from this file. Let me fix that and push a new change.

Comment (fabioy) [inline cluster/addons/dns/kube2sky/kube2sky.go]: Ok, taken care of. Thanks.

Comment (thockin) [inline cluster/addons/dns/kube2sky/kube2sky.go]: makes much more sense now :)

**Pull request merged**